public/app/plugins/datasource/mssql/response_parser.ts

```diff
@@ -1,73 +1,41 @@
-  target: string;
-  refId: string;
-  meta: any;
-  datapoints: [any[]];
-}
-
-export interface MssqlResponse {
-  data: Array<TableResponse | SeriesResponse>;
-}
+import { AnnotationEvent, DataFrame, FieldType, MetricFindValue } from '@grafana/data';
+import { BackendDataSourceResponse, toDataQueryResponse, FetchResponse } from '@grafana/runtime';
 
 export default class ResponseParser {
-  processQueryResult(res: any): MssqlResponse {
-    const data: any[] = [];
-
-    if (!res.data.results) {
-      return { data };
-    }
-
-    for (const key in res.data.results) {
-      const queryRes = res.data.results[key];
-
-      if (queryRes.series) {
-        for (const series of queryRes.series) {
-          data.push({
-            target: series.name,
-            datapoints: series.points,
-            refId: queryRes.refId,
-            meta: queryRes.meta,
-          });
-        }
-      }
-
-      if (queryRes.tables) {
-        for (const table of queryRes.tables) {
-          table.type = 'table';
-          table.refId = queryRes.refId;
-          table.meta = queryRes.meta;
-          data.push(table);
-        }
-      }
-    }
-
-    return { data: data };
-  }
+  transformMetricFindResponse(raw: FetchResponse<BackendDataSourceResponse>): MetricFindValue[] {
+    const frames = toDataQueryResponse(raw).data as DataFrame[];
 
-  parseMetricFindQueryResult(refId: string, results: any): MetricFindValue[] {
-    if (!results || results.data.length === 0 || results.data.results[refId].meta.rowCount === 0) {
+    if (!frames || !frames.length) {
       return [];
     }
 
-    const columns = results.data.results[refId].tables[0].columns;
-    const rows = results.data.results[refId].tables[0].rows;
-    const textColIndex = this.findColIndex(columns, '__text');
-    const valueColIndex = this.findColIndex(columns, '__value');
+    const frame = frames[0];
+
+    const values: MetricFindValue[] = [];
+    const textField = frame.fields.find((f) => f.name === '__text');
+    const valueField = frame.fields.find((f) => f.name === '__value');
 
-    if (columns.length === 2 && textColIndex !== -1 && valueColIndex !== -1) {
-      return this.transformToKeyValueList(rows, textColIndex, valueColIndex);
+    if (textField && valueField) {
+      for (let i = 0; i < textField.values.length; i++) {
+        values.push({ text: '' + textField.values.get(i), value: '' + valueField.values.get(i) });
+      }
+    } else {
+      const textFields = frame.fields.filter((f) => f.type === FieldType.string);
+      if (textFields) {
+        values.push(
+          ...textFields
+            .flatMap((f) => f.values.toArray())
+            .map((v) => ({
+              text: '' + v,
+            }))
+        );
+      }
     }
 
-    return this.transformToSimpleList(rows);
+    return Array.from(new Set(values.map((v) => v.text))).map((text) => {
+      return { text, value: values.find((v) => v.text === text)?.value } as MetricFindValue;
```

Comment:
Don't think there's a need for this cast?

Reply:
Good catch. Pushed fix